src/block/video-popup/index.php

```diff
@@ -19,3 +19,60 @@ function stackable_load_videopopup_frontend_script() {
 	}
 	add_action( 'stackable/video-popup/enqueue_scripts', 'stackable_load_videopopup_frontend_script' );
 }
+
+if ( ! class_exists( 'Stackable_Video_Popup_Schema' ) ) {
+	class Stackable_Video_Popup_Schema {
+		public $video_entities = [];
+
+		function __construct() {
+			add_filter( 'render_block_stackable/video-popup', array( $this, 'render_block_video_popup_schema' ), 10, 2 );
+			add_filter( 'wp_footer', array( $this, 'print_video_popup_schema' ) );
+		}
+
+		public function print_video_popup_schema() {
+			if ( count( $this->video_entities ) ) {
+				// Compile all video schema entities into a single script
+				echo '<script type="application/ld+json"> [ ' . implode( ', ', $this->video_entities ) . ' ] </script>';
+			}
+		}
+
+		public function render_block_video_popup_schema( $block_content, $block ) {
+			// Initialize video schema
+			$video_schema = array();
+			$video_schema[ '@context' ] = 'https://schema.org';
+			$video_schema[ '@type' ] = 'VideoObject';
+
+			// Get video schema properties from block attributes
+			$attributes = $block[ 'attrs' ];
+
+			// Get video name from the title of the post if not set
+			$name = isset( $attributes[ 'videoName' ] ) ? $attributes[ 'videoName' ] : ( get_the_title() ?? '');
+			// Get video upload date from the date of the post if not set
+			$upload_date = isset( $attributes[ 'videoUploadDate' ] ) ? $attributes[ 'videoUploadDate' ] : ( get_the_date( 'c' ) || '');
```

Comment:
Some errors on the date:
1. This is giving `true` as the value if no upload date was given.
2. If the date is given, it's showing an error:
<img width="615" alt="image" src="https://github.com/user-attachments/assets/ff699c56-08a6-4250-bee8-61832b8f3d76" />

Note: A passing value is of the format: `2025-04-10T03:24:47+00:00`